src/libstd/collections/hash/table.rs

```diff
@@ -835,18 +838,27 @@ pub struct IterMut<'a, K: 'a, V: 'a> {
     elems_left: usize,
 }
 
+unsafe impl<'a, K: Sync, V: Sync> Sync for IterMut<'a, K, V> {}
+unsafe impl<'a, K: Sync, V: Send> Send for IterMut<'a, K, V> {}
```

Comment:
@GuillaumeGomez As you can see this is tricky stuff! Can you change `IterMut` to use `K: Send` for its `Send` impl? I'd add a comment too, maybe like this:

```
// Both K: Sync and K: Send are correct for IterMut's Send impl, but Send is the more useful bound
```